### ISSUE TITLE
[Bug] [Android] CheckBox tab order

### ISSUE THREAD

**Issue by davidbritch** [xamarin/Xamarin.Forms#6955]
### Description

On Android, the tab order for the following controls is `Entry`, `CheckBox`, and then selection returns to the first `Entry`. The second is never selected.

```
<Entry />
<CheckBox />
<Entry />
```

### Steps to Reproduce

1. Run the attached sample.
2. Use the TAB key to move through the controls.

### Expected Behavior

Tab order is `Entry`, `CheckBox`, `Entry`.

### Actual Behavior

On Android, the tab order is `Entry`, `CheckBox` then selection returns to the first `Entry` rather than the second.

### Basic Information

- Version with issue: 4.1.0.581479
- Last known good version: N/A
- IDE: VSMac 8.1.5
- Platform Target Frameworks:
  - Android: API 28

### Reproduction Link

[test.zip](https://github.com/xamarin/Xamarin.Forms/files/3421908/test.zip)

**Comment by PureWeen** [comment]
Confirmed bug

It also doesn't work if you set the TabIndex yourself on the controls

**Comment by andreinitescu** [comment]
Can I suggest something. Instead of fixing CheckBox, why not make a decision quicker about my approach for RadioButton in  #5349 and if that's OK, just apply the same to CheckBox. It make sense for CheckBox and RadioButton have same approach.

**Comment by PureWeen** [comment]
I tested this on .NET MAUI and XF and it appears to be fixed

If you haven't already, I also recommend updating to the latest version Xamarin.Forms or at least to Xamarin.Forms 5.0.0.2083, which has many accessibility improvements. These improvements include an experimental accessibility flag that addresses a lot of improvements in Android accessibility.



To leverage this flag on the latest versions of Xamarin Forms, the following code can be inserted in the App.xaml.cs file, following the InitializeComponent call and before the MainPage is set: Xamarin.Forms.Device.SetFlags(new List<string> { "Accessibility_Experimental"});
